packages/circle-sdk-setup/src/cli.ts

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env node
+
+import { CircleSdk, RegisteredEntity, SecretApi } from 'web3-circle-sdk';
+import { Command } from 'commander';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as dotenv from 'dotenv';
+
+interface SetupOptions {
+  apiKey: string;
+  envPath?: string;
+}
+
+class CircleSdkSetup {
+  private options: Required<SetupOptions>;
+
+  constructor(options: SetupOptions) {
+    this.options = {
+      envPath: path.resolve(process.cwd(), '.env'),
+      ...options,
+    };
+  }
+
+  async checkExistingConfig(): Promise<boolean> {
+    console.log('Checking existing configuration...');
+    try {
+      if (fs.existsSync(this.options.envPath)) {
+        const envContent = dotenv.parse(fs.readFileSync(this.options.envPath));
+        return !!(envContent.CIRCLE_SECRET && envContent.CIRCLE_APP_ID);
+      }
+      return false;
+    } catch (error) {
+      console.error('Error checking existing configuration:', error);
+      return false;
+    }
+  }
+
+  async generateSecret(): Promise<string> {
+    console.log('Generating secret...');
+    const secret = SecretApi.generateSecret();
+    console.log('Secret generated:', secret);
+    return secret;
+  }
+
+  async initializeSdk(apiKey: string, secret: string): Promise<CircleSdk> {
+    console.log('Initializing Circle SDK...');
+    return new CircleSdk(apiKey, secret);
+  }
```

Comment:
Does this need to be async and return a `Promise`? Is this method even really needed?

Reply:
for context, that was my late night experiment with using AI agent to build something useful and I thought simple CLI tool is a good task for that 😄 . So I wasn't very strict with the results. I'm cleaning this up now, thanks for the review!